### PR TITLE
Add bitarray

### PR DESCRIPTION
**Pull request by hoodmane**
We need another trampoline patch for `cfunction_vectorcall_O`.

Thanks to @ilanschnell for his nice package!

### CODE REVIEW

File: cpython/patches/0001-Add-one-more-trampoline-in-cfunction_vectorcall_O.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
+From c454e4ebf79380fd206e3caf5cae343bc221b2cb Mon Sep 17 00:00:00 2001
```
Comment (hoodmane) [inline cpython/patches/0001-Add-one-more-trampoline-in-cfunction_vectorcall_O.patch:1]: Already fixed upstream.

File: packages/bitarray/patches/0002-skip-bad-tests-for-pyodide.patch [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+diff --git a/bitarray/test_bitarray.py b/bitarray/test_bitarray.py
+index e135528..6f45e60 100644
+--- a/bitarray/test_bitarray.py
++++ b/bitarray/test_bitarray.py
+@@ -3301,6 +3301,7 @@ class FileTests(unittest.TestCase, Util):
+             self.assertIsType(f, 'frozenbitarray')
+             self.check_obj(f)
+ 
++    @unittest.skip("not on pyodide")
```
Comment (hoodmane) [inline packages/bitarray/patches/0002-skip-bad-tests-for-pyodide.patch:18]: needs dbm

File: packages/bitarray/patches/0002-skip-bad-tests-for-pyodide.patch [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     def test_shelve(self):
+         if hasattr(sys, 'gettotalrefcount'):
+             return
+@@ -3547,6 +3548,7 @@ class FileTests(unittest.TestCase, Util):
+ 
+         self.assertEqual(self.read_file(), 1000 * b'\x55')
+ 
++    @unittest.skip("not on pyodide")
+     def test_mmap_2(self):
```
Comment (hoodmane) [inline packages/bitarray/patches/0002-skip-bad-tests-for-pyodide.patch:27]: hits emscripten mmap bug

**Comment by ilanschnell** [comment]
Great!  I'm happy to see bitarray being supported by pyodide!

**Comment by ilanschnell** [comment]
I've upstreamed skipping the two tests, so the next update (to bitarray 2.5.0) won't need the patch anymore.

**Comment by ilanschnell** [comment]
I've upstreamed skipping the two tests, so the next update (to bitarray 2.5.0) won't need the patch anymore.

**Review by ryanking13 — APPROVED** [review]
Yes, bitarray is a great package that we need to support. Thanks, @hoodmane, and thanks @ilanschnell for applying patches to the upstream!

**Comment by ryanking13** [comment]
Please also update the changelog

**Comment by hoodmane** [comment]
Thanks again @ilanschnell!

**Comment by ilanschnell** [comment]
Thanks for merging!  I'm happy to see bitarray being supported in pyodide!

**Pull request merged**